python/vmaf/tools/reader.py

```diff
@@ -190,27 +193,20 @@ def next(self, format='uint'):
 
         elif format == 'float':
             if self._is_8bit():
-                y = y.astype(np.double) / (2.0**8 - 1.0)
-                u = u.astype(np.double) / (2.0**8 - 1.0) if u is not None else None
-                v = v.astype(np.double) / (2.0**8 - 1.0) if v is not None else None
-                return y, u, v
+                bit_depth = 8
             elif self._is_10bitle():
-                y = y.astype(np.double) / (2.0**10 - 1.0)
-                u = u.astype(np.double) / (2.0**10 - 1.0) if u is not None else None
-                v = v.astype(np.double) / (2.0**10 - 1.0) if v is not None else None
-                return y, u, v
+                bit_depth = 10
             elif self._is_12bitle():
-                y = y.astype(np.double) / (2.0**12 - 1.0)
-                u = u.astype(np.double) / (2.0**12 - 1.0) if u is not None else None
-                v = v.astype(np.double) / (2.0**12 - 1.0) if v is not None else None
-                return y, u, v
+                bit_depth = 12
             elif self._is_16bitle():
-                y = y.astype(np.double) / (2.0**16 - 1.0)
-                u = u.astype(np.double) / (2.0**16 - 1.0) if u is not None else None
-                v = v.astype(np.double) / (2.0**16 - 1.0) if v is not None else None
-                return y, u, v
+                bit_depth = 16
             else:
                 assert False
 
+            y = self.convert_format(y, bit_depth)
+            u = self.convert_format(u, bit_depth)
+            v = self.convert_format(v, bit_depth)
```

Comment:
I think just changing this to
```
u = self.convert_format(u, bit_depth) if u is not None else None
v = self.convert_format(v, bit_depth) if v is not None else None
```
would maintain the existing behavior. No concerns other than this.

Reply:
Hello @nilfm, Thank you for your review! I have revised the code according to your instructions.

Reply:
Looks good to me! Tagging @li-zhi since he wrote most of this code, but I'll be merging soon if no objections.